services/command-router/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/KafkaBasedCommandConsumerFactoryImpl.java

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.commandrouter.impl.kafka;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.eclipse.hono.adapter.client.command.kafka.KafkaBasedInternalCommandSender;
+import org.eclipse.hono.client.kafka.HonoTopic;
+import org.eclipse.hono.client.kafka.KafkaProducerConfigProperties;
+import org.eclipse.hono.client.kafka.KafkaProducerFactory;
+import org.eclipse.hono.client.kafka.consumer.KafkaConsumerConfigProperties;
+import org.eclipse.hono.commandrouter.CommandConsumerFactory;
+import org.eclipse.hono.commandrouter.CommandTargetMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+
+/**
+ * A factory for creating clients for the <em>Kafka messaging infrastructure</em> to receive commands.
+ * <p>
+ * This factory uses wild-card based topic pattern to subscribe for commands, which receives the command 
+ * messages irrespective of the tenants.
+ * <p>
+ * Command messages are first received by the kafka based consumer. It is then determined which protocol adapter instance
+ * can handle the command. The command is then forwarded to the Kafka cluster on an topic containing that adapter
+ * instance id.
+ */
+public class KafkaBasedCommandConsumerFactoryImpl implements CommandConsumerFactory {
+    private static final Logger log = LoggerFactory.getLogger(KafkaBasedCommandConsumerFactoryImpl.class);
+    private static final Pattern topicPatternForIncomingCommands = Pattern
+            .compile(new HonoTopic(HonoTopic.Type.COMMAND, ".*").toString());
```

Comment:
Better use upper case for these constants. (Using `COMMANDS_TOPIC_PATTERN` could be a shorter alternative.)